plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -426,6 +427,16 @@ int process_event(struct flb_cloudwatch *ctx, struct cw_flush *buf,
             flb_plg_warn(ctx->ins, "[size=%zu] Truncating event which is larger than "
                          "max size allowed by CloudWatch", written);
             written = MAX_EVENT_LEN;
+
+            /* remove trailing unescaped backslash if inadvertently synthesized */
+            while (trailing_backslash_count < written &&
+                   buf->event_buf[(written - 1) - trailing_backslash_count] == '\\') {
+                trailing_backslash_count++;
+            }
+            if (trailing_backslash_count % 2 == 1) {
+                /* odd number of trailing backslashes, remove unpaired backslash */
+                written--;
+            }
```

Comment:
So looks like this code block is only inside the log_key if block above... is that intentional? this issue only happens if `log_key` is configured?